src/proj.rs

```diff
@@ -1086,15 +1087,18 @@ impl Proj {
         if let Some(indentation_width) = indentation_width {
             opts.push(format!("INDENTATION_WIDTH={}", indentation_width));
         }
+        if let Some(schema) = schema {
+            opts.push(format!("SCHEMA={}", schema));
+        }
         // Do we have input options? Join them into a single string
         let sep = if opts.len() > 1 { "," } else { "" };
         let joined = opts.join(sep);
         let opts_c = CString::new(joined)?;
-        let opts_p = if opts_c.is_empty() {
-            // You cannot pass an empty string as an input option: it must be a null pointer
-            vec![ptr::null()]
-        } else {
-            vec![opts_c.as_ptr()]
+        // it seems wasteful to leave the None check until now, but building the option pointers inside
+        // an if statement consistently causes libproj to segfault with a memory error due to bad input
```

Comment:
I thought... that sounds weird! Surely it must work in an if statement. And I pushed a commit and found that did not work 🙈

Reply:
That is so weird. Feel free to unwind/write over the last commit